Adding onboarding docs and configuring live preview (#5333)
ROCm/rocm-libraries: projects/hipdnn/docs/user-guides/how-to/build-execute-hipdnn.rst

  // Results available via yHostPtr[].

This is the basic frontend workflow:

1. Instantiate a :ref:`graph` that houses tensors and operations.
2. Create input tensors for the operations within the graph.
3. Add operations, which become :ref:`nodes`, attaching the input tensors to the nodes and creating output tensors from the node's operation. Any :ref:`attributes` you add configure the behavior of these nodes.
4. Continue adding operations and attributes using the output tensors from prior nodes as input tensors to new nodes.
5. The graph is processed to find a matching engine.
6. (Optional) Any non-default engine-specific configuration knobs are applied.
7. Execution plans are built, memory is allocated, tensor data is initialized.
8. The resulting plan is executed on the GPU hardware.

For complete working examples, see the official `samples on GitHub <https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipdnn/samples>`_.

Variant pack
------------

The variant pack specifies which GPU memory address correspond to which tensors. 
The tensors are added to the variant pack using the tensor UID as the key and the GPU memory address as the value. 

> It's the responsibility of the application to ensure sufficient GPU memory is allocated for each tensor, and to ensure data is properly to/from the GPU and host memory. 

Reviewer: ```suggestion
It's the responsibility of the application to ensure sufficient GPU memory is allocated for each tensor, and to ensure data is synchronized between the GPU and host memory. 
```

Author: added

.. meta::
  :description: Learn how to build and execute operation graphs in hipDNN.
  :keywords: hipDNN, ROCm, graphs

.. _build-execute:

> 

Reviewer 2: Is it on purpose that the How To topics include Windows but not all topics do?

<img width="571" height="315" alt="Image" src="https://github.com/user-attachments/assets/9def676c-7fa0-4a89-af5a-cac8fdd33978" />

Author: Fixed this issue

********************************************
Build and execute operation graphs in hipDNN
********************************************

> This section covers how to use the frontend API to build and execute graph operations in hipDNN.

Reviewer 2: ```suggestion
This topic covers how to use the frontend API to build and execute graph operations in hipDNN.
```

> 

Reviewer 2: Does the intro here need to discuss that this is for users (that role)? The distinction is made in the TOC.

Reviewer 2: I won't mark up the other topics with the same comment.

Author: It's primarily for users, but plugin developers can use these topics too. The articles specifically for the developers only apply to them.

The hipDNN frontend provides a C++ header-only API for building and executing operation graphs.

Frontend file structure
=======================

Here's the basic frontend file structure with links to the GitHub repository:

- `Library includes <https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipdnn/frontend/include>`_
- `Samples <https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipdnn/samples>`_
- `Unit tests <https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipdnn/frontend/tests>`_

Frontend architecture
=====================

See :ref:`architecture` for a conceptual description of the hipDNN graph, tensors, nodes, and attributes.

See :ref:`plugin-support` for a detailed list of the supported operations.

Simplified workflow example
===========================

> This example demonstrates sample code that creates a graph, creates tensors, and adds operations before building and executing them. Here's a simplified workflow example:

Reviewer 2: Something off in this sentence. 
This example doesn't demonstrate sample code. 

Perhaps:
The following sample code demonstrates...

Author: resolved

.. code:: cpp

  // Create a graph
  Graph graph;
  graph.set_compute_data_type(DataType_t::FLOAT);

  // Create tensors
  auto x = Graph::tensor(/* tensor attributes */);
  auto scale = Graph::tensor(/* tensor attributes */);
  auto bias = Graph::tensor(/* tensor attributes */);

  // Add operations
  auto [y, mean, inv_var, _, _] = graph.batchnorm(x, scale, bias, bn_attributes);

  // Build and execute
  graph.build_operation_graph(handle);
  graph.create_execution_plans();
  graph.build_plans();
  graph.execute(handle, variant_pack, workspace);

This is the basic frontend workflow:

1. Instantiate a :ref:`graph` that houses tensors and operations.
2. Create input tensors for the operations within the graph.
3. Add operations, which become :ref:`nodes`, attaching the input tensors to the nodes and creating output tensors from the node's operation. Any :ref:`attributes` you add configure the behavior of these nodes.

> 4. Continue adding operations and attributes using the output tensors from prior nodes as input tensors for new nodes.

Reviewer 2: ```suggestion
4. Continue adding operations and attributes using the output tensors from prior nodes as input tensors to new nodes.
```

Reviewer 2: Not sure where it's **to** or **for**
